launch_ros/launch_ros/substitutions/parameter.py

```diff
@@ -38,17 +42,41 @@ class Parameter(Substitution):
     def __init__(
         self,
         name: SomeSubstitutionsType,
+        *,
+        default: Optional[Union[Any, Iterable[Any]]] = None
     ) -> None:
         """Create a Parameter substitution."""
         super().__init__()
         self.__name = normalize_to_list_of_substitutions(name)
+        if default is None:
+            self.__default = default
+        else:
+            # convert any items in default that are not a Substitution or str to a str
+            str_normalized_default = []  # type: List[Union[Text, Substitution]]
+            definitely_iterable_default = ((),)  # type: Iterable[Any]
+            if isinstance(default, collections.abc.Iterable):
+                definitely_iterable_default = default
+            else:
+                definitely_iterable_default = (default,)
+            for item in definitely_iterable_default:
+                if isinstance(item, (str, Substitution)):
+                    str_normalized_default.append(item)
+                else:
+                    str_normalized_default.append(str(item))
+            # use normalize_to_list_of_substitutions to convert str to TextSubstitution's too
+            self.__default = \
+                normalize_to_list_of_substitutions(
+                    str_normalized_default)  # type: List[Substitution]
```

Comment:
Is this type annotation useful since `self.__default` may be `None` in some cases? Anyways, [`normalize_to_list_of_substitutions`](https://github.com/ros2/launch/blob/d2935d6fd85636ace62699eec106cd56f5e9c6a4/launch/launch/utilities/normalize_to_list_of_substitutions_impl.py#L26) already contains a return type annotation, so think the comment is not needed.

```suggestion
            self.__default = normalize_to_list_of_substitutions(str_normalized_default)
```

Reply:
Actually, I just copied it from `launch_configuration.py`.
Removed in 9e6478809d770e62cefd494a3317ea2168ec0ad6.